docs/manual/annotating-libraries.tex

```diff
@@ -536,8 +536,12 @@
 a warning whenever a stub file specification is redundant with
 bytecode annotations.
 
-A file being compiled takes precedence over a stub file.  If file \<A.java>
-is being compiled, then any stub for class \<A> is ignored.
+By default, a file being compiled takes precedence over a stub file:  if
+file \<A.java> is being compiled, then any stub for class \<A> is ignored.
+The command-line option \<-AmergeStubsWithSource> tells the checker to
+take the greatest lower bound of the type in the stub file and the type in
```

Comment:
I'm not sure that using the greatest lower bound will be the most intuitive behavior.
Let's take source code that is annotated `@A` and a stub that is annotated `@B` and they are not comparable. Taking the GLB means using some other annotation that neither location had specified.
How about limiting this to specifying subtypes (or uselessly supertypes) in stubfiles? That would make it easier to deduce where information comes from.

Reply:
I see your point. That's a pretty straightforward change, and it shouldn't impact my use case, so I'll go ahead and make it. Thanks for the feedback.

Reply:
It's actually not as straightforward as I had thought. Changing this to only select the stub type if it is a stub type causes the WPI tests to fail when using the `-Ainfer=stubs` feature, because one of those tests *requires* that the output of WPI and an annotation written in source code be merged (`TypeVariableTest3.java`).

I would prefer to match the existing WPI behavior and merge the types. I know I'm making the decision for the benefit of WPI here rather than a hypothetical other use case. That said, I don't think this is confusing enough to warrant changing WPI's behavior or letting the jaif and stub versions diverge.

Reply:
Thanks for checking. I'm fine with using GLB. maybe some additional comment in the doc would be useful.

Reply:
I'll add a note. Thanks!

Reply:
I don't see a note in the documentation about the choice of GLB being for the convenience of WPI.